bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\Date\Value;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\HandlerType;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+
+use function in_array;
+
+final class ScheduledVisibilityService
+{
+    public function __construct(
+        private readonly iterable $handlers,
+        private readonly string $type,
+        private readonly bool $enabled,
+        private readonly bool $allContentTypes,
+        private readonly array $allowedContentTypes,
+    ) {}
+
+    public function toggleVisibility(Content $content): void
+    {
+        $handlerType = HandlerType::from($this->type);
+
+        /** @var ScheduledVisibilityInterface $handler */
+        foreach ($this->handlers as $handler) {
+            if ($handler->getType() === $handlerType) {
+                if ($this->shouldHide($content)) {
+                    $handler->hide($content);
+
+                    return;
+                }
+                if ($this->shouldReveal($content)) {
+                    $handler->reveal($content);
+                }
+
+                return;
+            }
+        }
+    }
+
+    public function accept(Content $content): bool
+    {
+        $enabled = $this->enabled;
+        if (!$enabled) {
+            return false;
+        }
+
+        $allowedAll = $this->allContentTypes;
+        $allowedContentTypes = $this->allowedContentTypes;
+        $contentType = $content->getContentType();
+        if (!$allowedAll && !in_array($contentType->identifier, $allowedContentTypes, true)) {
+            return false;
+        }
+
+        $fieldDefinitions = $contentType->getFieldDefinitions();
+        if (!$fieldDefinitions->has('publish_from') || !$fieldDefinitions->has('publish_to')) {
+            return false;
+        }
+
+        $publishFromField = $content->getField('publish_from');
+        $publishToField = $content->getField('publish_to');
```

Comment:
`Content::getField()` method can return `null` if the field is missing in the Type, we should check whether that's the case.

Reply:
wouldn't that be covered by this code few lines above?
`        $fieldDefinitions = $contentType->getFieldDefinitions();
        if (!$fieldDefinitions->has('publish_from') || !$fieldDefinitions->has('publish_to')) {
            return false;
        }`

Reply:
Yes it would, my bad. But then add inline type hints that the fetched fields are instances of `Field` and not null.

Reply:
fixed in 1e6dd6e44fcdde4ee48014e9ba7dae6be86e2292